addons/hr_expense/tests/test_expenses.py

```diff
@@ -350,3 +350,63 @@ def test_print_expense_check(self):
             'amount_paid': formatLang(self.env, 11.0, currency_obj=self.env.company.currency_id),
             'currency': self.env.company.currency_id
         })
+
+    def test_reset_move_to_draft(self):
+        """
+        Test the state of an expense and its report
+        after resetting the paid move to draft
+        """
+        # Create expense and report
+        expense = self.env['hr.expense'].create({
+            'name': 'expense_1',
+            'employee_id': self.expense_employee.id,
+            'product_id': self.product_a.id,
+            'unit_amount': 1000.00,
+        })
+        expense.action_submit_expenses()
+        expense_sheet = expense.sheet_id
+
+        self.assertEqual(expense.state, 'draft', 'Expense state must be draft before sheet submission')
+        self.assertEqual(expense_sheet.state, 'draft', 'Sheet state must be draft before submission')
+
+        # Submit report
+        expense_sheet.action_submit_sheet()
+
+        self.assertEqual(expense.state, 'reported', 'Expense state must be reported after sheet submission')
+        self.assertEqual(expense_sheet.state, 'submit', 'Sheet state must be submit after submission')
+
+        # Approve report
+        expense_sheet.approve_expense_sheets()
+
+        self.assertEqual(expense.state, 'approved', 'Expense state must be draft after sheet approval')
+        self.assertEqual(expense_sheet.state, 'approve', 'Sheet state must be draft after approval')
+
+        # Create move
+        res = expense_sheet.action_sheet_move_create()
+
+        self.assertEqual(expense.state, 'approved', 'Expense state must be draft after posting move')
+        self.assertEqual(expense_sheet.state, 'post', 'Sheet state must be draft after posting move')
+
+        # Pay move
+        move = self.env['account.move'].browse(res[expense_sheet.id].id)
```

Comment:
Why browse `account.move` when you already have the record with `res[expense_sheet.id]`?

Reply:
I could do like here https://github.com/odoo/odoo/blob/5c74a71b53b83630a8d4eb1187aacb3d7fd92b91/addons/hr_expense/tests/test_expenses.py#L292-L294
but as I need the move later, I believe it is ok to browse it here and pay it directly, what do you think?